rest/cloudevents/cloudEvents.go

```diff
@@ -37,6 +38,14 @@ func (sv SpecVersion) IsValid() error {
 	return fmt.Errorf("invalid cloud events spec version, expect %v, got %v", V102, sv)
 }
 
+func SourceIsValid(uri string) error {
```

Comment:
I would like this to be attached to the data type. You will have create a new one for the source which is currently just a string.

The interface should be similar to the content type and version.